roles/pulp/README.md

```diff
@@ -26,9 +29,12 @@ Role Variables:
     See `prereq_pip_packages` also.
 * `pulp_install_api_service`: Whether to create systemd service files for
   pulpcore-api. Defaults to "true".
+* `pulp_update`: Whether to update/upgrade pulpcore to the latest stable release from PyPI within `pulp_branch`. Only affects systems where Pulp is already installed. To limit this to micro (z-stream) updates, make sure to set `pulp_branch`. If `pulpcore_version` is set, or `pulp_source_dir` is set, this has no effect and is effectively always `true`. Defaults to "false".
+* `pulp_branch`: The branch of of pulpcore (or update/upgrade to the latest release from, see `pulp_update`). Defaults to the latest stable branch release of pulpcore at the time of ansible-pulp being released, which is currently `3.0`. It is recommended to set this if you ever plan to re-run the ansible installer; when a new branch is released and if you update ansible-pulp, some of your content plugins may not be compatible yet. An example value is `3.1`. Ignored if `pulp_source_dir` or `pulp_version` is set. Do not set this to a version higher than the current default (unless you are installing a development branch of pulpcore, and have updated ansible-pulp 1st.)
+* `pulp_version`: Optional. A specific version of pulp to install from PyPI. Ignored if `pulp_source_dir` is set. Overrides `pulp_update` to `true`.
```

Comment:
Thank you for the review!

I didn't plan to write any code to override it. Rather, I am mostly certain that the override is a limitation of ansible's pip module, which is a wrapper around the `pip` command. And I'd rather not write & maintain a ton of code to workaround it ("embrace the constraint.")

I assume that calling the ansible pip module with `version` (which is only [documented](https://docs.ansible.com/ansible/latest/modules/pip_module.html) as accepting values like `3.0.1`, not `>=3.0.0`) maps to it calling `pip install pulpcore==3.0.1`. With `version` undefined, I assume it maps it to calling `pip install pulpcore`.

I know ([because I looked at the code](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/language/pip.py) that calling the ansible pip module with `state=latest` maps to `pip install --upgrade`, whereas `state=present` maps to `pip install`.

However the pip command, --upgrade is ignored whenever the exact version is specified. It will upgrade or downgrade regardless. I just tested this to be certain:

```
(upgrade-test) [mdepaulo@mdepaulo upgrade-test]$ pip install --upgrade pytest==5.3.1
...
Successfully installed pytest-5.3.1


(upgrade-test) [mdepaulo@mdepaulo upgrade-test]$ pip install --upgrade pytest==5.3.2
...
Successfully installed pytest-5.3.2


(upgrade-test) [mdepaulo@mdepaulo upgrade-test]$ pip install --upgrade pytest==5.3.1
...
Successfully installed pytest-5.3.1


(upgrade-test) [mdepaulo@mdepaulo upgrade-test]$ pip install pytest==5.3.2
...
Successfully installed pytest-5.3.2


(upgrade-test) [mdepaulo@mdepaulo upgrade-test]$ pip install pytest==5.3.1
...
Successfully installed pytest-5.3.1

```

Reply:
Maybe it is not documented, but i tried to pass a non exact version to it, and it didn't report an error. I guess we still need to verify this also works as expected, but '<3.1' with or without '--upgrade' should be two different things.

Reply:
The examples suggest that it is possible to have complex version specifiers:
https://docs.ansible.com/ansible/latest/modules/pip_module.html#examples